src/applications/vaos/appointment-list/components/AppointmentsPageV2/Grid.jsx

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {
+  getAppointmentDate,
+  getAppointmentTimezone,
+  isClinicVideoAppointment,
+  isVAPhoneAppointment,
+} from '../../../services/appointment';
+import { APPOINTMENT_STATUS, VIDEO_TYPES } from '../../../utils/constants';
+import { getTypeOfCareById } from '../../../utils/appointment';
+import { Label } from './Label';
+// import { handleClick, handleKeyDown, Label } from './AppointmentListItem';
+
+function isInPersonVAAppointment(appointment) {
+  const { isCommunityCare, isVideo } = appointment?.vaos || {};
+  const isPhone = isVAPhoneAppointment(appointment);
+
+  return !isVideo && !isCommunityCare && !isPhone;
+}
+
+function getVideoAppointmentLocationText(appointment) {
+  const { isAtlas } = appointment.videoData;
+  const videoKind = appointment.videoData.kind;
+  let desc = 'Video appointment at home';
+
+  if (isAtlas) {
+    desc = 'Video appointment at an ATLAS location';
+  } else if (isClinicVideoAppointment(appointment)) {
+    desc = 'Video appointment at a VA location';
+  } else if (videoKind === VIDEO_TYPES.gfe) {
+    desc = 'Video with VA device';
+  }
+
+  return desc;
+}
+
+function getPractitionerName(appointment) {
+  const { practitioners } = appointment;
+
+  if (!practitioners) return '<name here>';
```

Comment:
adding the length will test against empty array
```suggestion
  if (!practitioners?.length) return '<name here>';
```

Reply:
Good catch!